apps/speed.c

```diff
@@ -169,11 +169,6 @@ static int DES_ede3_cbc_encrypt_loop(void *args);
 static int AES_cbc_128_encrypt_loop(void *args);
 static int AES_cbc_192_encrypt_loop(void *args);
 static int AES_cbc_256_encrypt_loop(void *args);
-#if !OPENSSL_API_3
```

Comment:
I don't get why these changes are included in this PR - they are actually unrelated to the adjustment of the option handling IMHO.

Reply:
It starts with this change in `include/openssl/aes.h`:

``` diff
@@ -73,17 +73,18 @@ void AES_cfb8_encrypt(const unsigned char *in, unsigned char *out,
 void AES_ofb128_encrypt(const unsigned char *in, unsigned char *out,
                         size_t length, const AES_KEY *key,
                         unsigned char *ivec, int *num);
-# if !OPENSSL_API_3
 /* NB: the IV is _two_ blocks long */
-void AES_ige_encrypt(const unsigned char *in, unsigned char *out,
-                     size_t length, const AES_KEY *key,
-                     unsigned char *ivec, const int enc);
+DEPRECATEDIN_3_0(void AES_ige_encrypt(const unsigned char *in,
+                                      unsigned char *out,
+                                      size_t length, const AES_KEY *key,
+                                      unsigned char *ivec, const int enc))
 /* NB: the IV is _four_ blocks long */
-void AES_bi_ige_encrypt(const unsigned char *in, unsigned char *out,
-                        size_t length, const AES_KEY *key,
-                        const AES_KEY *key2, const unsigned char *ivec,
-                        const int enc);
-# endif
+DEPRECATEDIN_3_0(void AES_bi_ige_encrypt(const unsigned char *in,
+                                         unsigned char *out,
+                                         size_t length, const AES_KEY *key,
+                                         const AES_KEY *key2,
+                                         const unsigned char *ivec,
+                                         const int enc))
 
 int AES_wrap_key(AES_KEY *key, const unsigned char *iv,
                  unsigned char *out,
```

This turn a "simulated deprecation" to an actual one (i.e. we not have a proper `deprecated` attribute attached to those functions with gcc and clang).

The consequence, though, is that any `--strict-warning` will make the build fail because the warning for using deprecated functions is turned into an error.  That's why I need to make that removal.  This is, btw, our M.O. so far, to remove all uses of deprecated functions, but leave the implementation around in case someone still wants to use them.

The alternative is to keep this as a "simulated deprecation" in this PR and handle the "real" deprecation of AES_ige functions in a separate PR.  I can certainly do that if that helps reviewing this PR.

Reply:
So...  was that explanation enough justification, or would you prefer I make a separate PR for this part?

Reply:
I'd be happier with them in a separate PR as they basically don't match the PR description (which was nicely worded and easy to follow). They just popped out as a "surprise" ...

Reply:
Okie.  Will do.

Reply:
Done.  I hope I didn't miss anything.